src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.File;
+
+/**
+ * The Storage class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor for a Storage object.
+     * @param filePath filePath of the file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the data of the file given and returns it in an ArrayList.
+     * @return An ArrayList of tasks stored in the file.
+     * @throws DukeException if file is not found.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+        try {
+            File file = new File(filePath);
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String str = sc.nextLine();
+                String[] command = str.split(" \\| ");
+                Task task;
+
+                switch (command[0]) {
+                    case "T":
+                        task = new ToDo(command[2]);
+                        break;
+                    case "D":
+                        task = new Deadline(command[2], LocalDate.parse(command[3], 
```

Comment:
Perhaps you should try to avoid deep nesting, as this line is 5 indentations in and you should try avoiding more than 3 indentations?

Reply:
Alright thanks